lib/src/widgets/chat/audio/ds_audio_message_bubble.dart

```diff
@@ -0,0 +1,221 @@
+import 'package:blip_ds/blip_ds.dart';
+import 'package:blip_ds/src/widgets/chat/audio/audio_seek_bar.widget.dart';
+import 'package:blip_ds/src/controllers/chat/ds_audio_message_bubble.controller.dart';
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+import 'package:just_audio/just_audio.dart';
+import 'package:flutter_spinkit/flutter_spinkit.dart';
+
+class DSAudioMessageBubble extends StatefulWidget {
+  final String uri;
+  final DSAlign align;
+  final List<DSBorderRadius> borderRadius;
+
+  const DSAudioMessageBubble({
+    Key? key,
+    required this.uri,
+    required this.align,
+    this.borderRadius = const [DSBorderRadius.all],
+  }) : super(key: key);
+
+  @override
+  State<DSAudioMessageBubble> createState() => _DSAudioMessageBubbleState();
+}
+
+class _DSAudioMessageBubbleState extends State<DSAudioMessageBubble>
+    with WidgetsBindingObserver {
+  final controller = DSAudioMessageBubbleController();
+
+  @override
+  void initState() {
+    super.initState();
+
+    init();
+  }
+
+  @override
+  void dispose() {
+    controller.player.dispose();
+    controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    if (state == AppLifecycleState.paused) {
+      controller.player.stop();
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return DSMessageBubble(
+      contentPadding: EdgeInsets.zero,
+      borderRadius: widget.borderRadius,
+      align: widget.align,
+      child: SizedBox(
+        height: 62.0,
+        child: Stack(
+          children: [
+            controlButtons(),
+            seekBar(),
+            audioSpeed(),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Future<void> init() async {
+    controller.player.playerStateStream.listen((event) {
+      if (event.processingState == ProcessingState.completed) {
+        controller.player.seek(Duration.zero);
+        controller.player.stop();
+      }
+    });
+
+    await controller.player
+        .setAudioSource(AudioSource.uri(Uri.parse(widget.uri)));
+  }
+
+  Widget controlButtons() {
```

Comment:
I think that every button inside this method should be a widget, mainly for two reasons: first, they could be reused, and second, this would improve the widget tree readability, since the Flutter framework is unaware of functions.

Detailed Explanation: https://stackoverflow.com/questions/53234825/what-is-the-difference-between-functions-and-classes-to-create-reusable-widgets